### PR TITLE
ROX-26300: Increase pagination limit to try to get more unique autocomplete results

### PR DESCRIPTION
**Pull request by charmik-redhat**
### Description

The autocomplete searcher first queries the database to get 10 matches, then removed duplicates from those matches. If there are too many duplicates, this results in autocomplete ultimately returning fewer matches. This PR increases the number of results queried from db to 100. This is chosen as a reasonably large limit to increase the chances of returning 10 suggestions after removing duplicates. If we still find a lot of duplicates, then we just return what we have. This may not be the best solution but we also want to return in a reasonable time.

### User-facing documentation

- [X] CHANGELOG update is not needed
- [X] documentation  is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [x] CI results are inspected

#### Automated testing

- [X] no unit tests added

Existing unit tests for the search service should be sufficient for this trivial change.

#### How I validated my change

Manual testing

### CODE REVIEW

File: central/search/service/service_impl.go [hunk 1]
```diff
@@ -195,84 +194,94 @@ func RunAutoComplete(ctx context.Context, queryString string, categories []v1.Se
...
-				utils.Should(errors.Errorf("searchers map has an entry for category %v, but the returned searcher was nil", category))
-			}
-			return nil, errors.Wrapf(errox.InvalidArgs, "Search category %q is not implemented", category.String())
-		}
 
-		optMultiMap := categoryToOptionsMultimap[category]
-		if optMultiMap == nil {
-			return nil, errors.Wrapf(errox.InvalidArgs, "Search category %q is not implemented", category.String())
+	resultSet := set.NewStringSet()
+	for i := 0; i < maxRetries; i++ {
```
Comment (SimonBaeumer) [inline central/search/service/service_impl.go:205]: How I understand this change is that we fetch the auto-complete result (max 10), remove all duplicates and may be left with fewer results. Wouldn't it be easier to just fetch more results and return max 10 by reducing the result set to 10 in Go code?

nit: Could you document why the retry is necessary?

**Review by SimonBaeumer — COMMENTED** [review]
@charmik-redhat for my understanding: Is there a doc or md file which describes on a high-level how auto-complete works?

File: central/search/service/service_impl.go (hunk 1 above)
Comment (dashrews78) [reply central/search/service/service_impl.go:205]: grab 100 and then reduce to 10.  makes some sense.  

Looks like there may be some mechanisms to get distinct on arrays and the jsonb objects that we use to store the maps.  Might be worth looking at as that would solve it entirely

Comment (dashrews78) [reply central/search/service/service_impl.go:205]: got it.  Just grabbing 100 or so seems to be the simplest path then.  That is a small enough number that it should not cause any issues.

Would it be worth writing a spike to ROX-24279?  Probably wouldn't be a huge priority, but worth documenting yet another limitation to how our framework builds queries.

Comment (charmik-redhat) [reply central/search/service/service_impl.go]: Changed it to just grab 100 results. And created [ROX-26368](https://issues.redhat.com/browse/ROX-26368)

File: central/search/service/service_impl.go [hunk 2]
```diff
@@ -42,7 +42,10 @@ import (
 	"google.golang.org/grpc"
 )
 
-const maxAutocompleteResults = 10
+const (
+	pagination             = 100
```
[the remark this answers is not in the document]
Comment (charmik-redhat) [reply central/search/service/service_impl.go]: Good suggestion

**Pull request merged**